### DISCUSSION TITLE
useEntityRecords tutorial

### DISCUSSION THREAD

**Discussion by mburridge** [WordPress/developer-blog-content#60] [Topic Ideas]
A tutorial explaining the use of the [`useEntityRecords` hook](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-core-data/#useentityrecords) to retrieve data from the database, and demonstrating its usage via a demo project. It will also show:

- how to use the state properties, e.g. `isResolving`, `hasResolved`, to carry out actions depending on the state of the request
- the use of attributes to alter the request query

The tutorial could also cover the [`useEntityRecord` hook](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-core-data/#useentityrecord). 

There are currently no tutorials that I can find and the documentation on this is scant.

**Comment by mburridge** [comment]
I would like to author this post, so I haven't added the `needs writer` label.

**Comment by ryanwelcher** [comment]
Love this idea! 

What would be awesome is a listing of all of the parameters that it can accept. It's not a 1:1 relationship to WP_Query args so having a list would be fantastic.

**Comment by bph** [comment]
Discussion is locked. Please add further comment to the issue 

- #87

**Comment by bph** [comment]
Published [useEntityRecords: an easier way to fetch WordPress data](https://developer.wordpress.org/news/2023/05/useentityrecords-an-easier-way-to-fetch-wordpress-data/)
